lib/id3c/api/routes.py

```diff
@@ -283,6 +285,147 @@ def get_identifier_set_uses(*, session):
 
     return jsonify([ use._asdict() for use in uses ])
 
+@api_v1.route("/warehouse/sample", methods = ['POST'])
+@content_types_accepted(["application/json"])
+@check_content_length
+@authenticated_datastore_session_required
+def post_sample(*, session):
+    """
+    Create/update a single sample.
+
+    POST /warehouse/sample to validate, then create new or update
+    existing sample that passes validation.
+
+    JSON object with status (inserted, updated, or failed validation)
+    of the sample is returned.
+    """
+
+    sample = request.get_json()
+
+    LOG.debug(f"Creating/updating sample «{sample}»")
+
+    # everything except _provenance, which is auto-calculated and added below
+    schema = {
```

Comment:
Should we put this into some sort of config file so the API code doesn't need to change if the database does?

Reply:
Good idea! This JSON Schema doesn't match the database exactly, it's more to enforce how we want to receive the data (flat JSON) but then its structure gets modified before it's stored. It does clutter things up a bit to include the schema here though, so moving it to a new file would keep things more organized.  I'll add a new folder `lib/id3c/api/schemas` and load them from there.